src/browser/setupWorker/start/createResponseListener.ts

```diff
@@ -25,10 +25,14 @@ export function createResponseListener(context: SetupWorkerInternalContext) {
       return
     }
 
-    const response =
-      responseJson.status === 0
-        ? Response.error()
-        : new Response(responseJson.body, responseJson)
+    let response: Response
```

Comment:
But yes, overall, I prefer this being a standalone method that we can also unit test to perfection.

Reply:
I think this style inline is probably the easiest to read, but not necessarily a big deal either way. 

Since we don't do this anywhere else, extracting it seems heavy, and more annoying to skim through, but no strong feeling